lti_consumer/lti_1p3/extensions/rest_framework/serializers.py

```diff
@@ -374,3 +376,89 @@ class LtiDlImageSerializer(serializers.Serializer):
     thumbnail = LtiDLIconPropertySerializer(required=False)
     width = serializers.IntegerField(min_value=1, required=False)
     height = serializers.IntegerField(min_value=1, required=False)
+
+
+class LtiContextSerializer(serializers.Serializer):
+    """
+    Serializer for a LTI Context
+    """
+    id = UsageKeyField()
+
+
+class LtiNrpsContextMemberBasicSerializer(serializers.Serializer):
+    """
+    Non PII fields serializer for Context Member.
+    """
+    status = serializers.SerializerMethodField()
+    user_id = serializers.SerializerMethodField()
+    roles = serializers.SerializerMethodField()
+
+    def get_user_id(self, user):
+        """
+        Get external user id for the user.
+        """
+        externalid, _ = compat.get_or_create_externalid(user)
+        return serializers.UUIDField().to_representation(externalid.external_user_id)
+
+    def get_status(self, user):
+        """
+        Get user status string value.
+        """
+        return 'Active' if user.is_active else 'Inactive'
+
+    def get_roles(self, user):
+        """
+        Prepare and return Context Roles for user.
+        """
+        roles = []
+        if user.courseenrollment_set.count() > 0:
+            roles += LTI_1P3_CONTEXT_ROLE_MAP['student']
+
+        for accessrole in user.courseaccessrole_set.all():
+            if LTI_1P3_CONTEXT_ROLE_MAP.get(accessrole.role):
+                roles += LTI_1P3_CONTEXT_ROLE_MAP[accessrole.role]
+
+        return set(roles)
+
+
+class LtiNrpsContextMemberPIISerializer(LtiNrpsContextMemberBasicSerializer):
+    """
+    Personally identifiable information serializer for Context Member.
+    """
+    name = serializers.CharField(source='profile.name', required=False)
```

Comment:
@shimulch You're not prefetching the `profile` user model here. This will cause a query explosion, you need to prefetch the profile to avoid performance issues on big courses.

Reply:
@giovannicimolin, since the data API now prefaces `profile`, this won't be a problem.